diff --git a/node_modules/ipfs-http-client/src/lib/core.js b/node_modules/ipfs-http-client/src/lib/core.js
index fb6b51e..0cd9806 100644
--- a/node_modules/ipfs-http-client/src/lib/core.js
+++ b/node_modules/ipfs-http-client/src/lib/core.js
@@ -25,7 +25,7 @@ const normalizeInput = (options = {}) => {
     options = { url: options }
   }
 
-  const url = new URL(options.url)
+  const url = new URL(options.url ?? options)
   if (options.apiPath) {
     url.pathname = options.apiPath
   } else if (url.pathname === '/' || url.pathname === undefined) {
