diff --git a/node_modules/iso-url/src/url-browser.js b/node_modules/iso-url/src/url-browser.js
index 2fb48b4..b4c9d83 100644
--- a/node_modules/iso-url/src/url-browser.js
+++ b/node_modules/iso-url/src/url-browser.js
@@ -1,12 +1,11 @@
 'use strict';
 
-const defaultBase = self.location ?
-    self.location.protocol + '//' + self.location.host :
-    '';
+const defaultBase = self.location && self.location.protocol + '//' + self.location.host;
 const URL = self.URL;
 
 class URLWithLegacySupport {
     constructor(url = '', base = defaultBase) {
+        console.log('iso-url constructor', { url, defaultBase });
         this.super = new URL(url, base);
         this.path = this.pathname + this.search;
         this.auth =
