diff --git a/node_modules/react-native/Libraries/Blob/FileReader.js b/node_modules/react-native/Libraries/Blob/FileReader.js
index 15a8b44..a407cff 100644
--- a/node_modules/react-native/Libraries/Blob/FileReader.js
+++ b/node_modules/react-native/Libraries/Blob/FileReader.js
@@ -81,8 +81,41 @@ class FileReader extends (EventTarget(...READER_EVENTS): any) {
     }
   }
 
-  readAsArrayBuffer() {
-    throw new Error('FileReader.readAsArrayBuffer is not implemented');
+  readAsArrayBuffer(blob: ?Blob) {
+    // throw new Error('FileReader.readAsArrayBuffer is not implemented');
+    this._aborted = false;
+
+    if (blob == null) {
+      throw new TypeError(
+        "Failed to execute 'readAsArrayBuffer' on 'FileReader': parameter 1 is not of type 'Blob'",
+      );
+    }
+
+    NativeFileReaderModule.readAsDataURL(blob.data).then(
+      (text: string) => {
+        if (this._aborted) {
+          return;
+        }
+
+        const bytes = atob(text.split(',')[1]);
+        const buffer = new ArrayBuffer(bytes.length);
+        const view = new Uint8Array(buffer);
+
+        for (let i = 0; i < bytes.length; i++) {
+          view[i] = bytes.charCodeAt(i);
+        }
+
+        this._result = view.buffer;
+        this._setReadyState(DONE);
+      },
+      error => {
+        if (this._aborted) {
+          return;
+        }
+        this._error = error;
+        this._setReadyState(DONE);
+      },
+    );
   }
 
   readAsDataURL(blob: ?Blob) {
@@ -126,6 +159,9 @@ class FileReader extends (EventTarget(...READER_EVENTS): any) {
         if (this._aborted) {
           return;
         }
+
+        console.log('ReactNative FileReader NativeFileReaderModule readAsText', { text });
+
         this._result = text;
         this._setReadyState(DONE);
       },
diff --git a/node_modules/react-native/Libraries/Network/FormData.js b/node_modules/react-native/Libraries/Network/FormData.js
index bdb1da3..5f3c852 100644
--- a/node_modules/react-native/Libraries/Network/FormData.js
+++ b/node_modules/react-native/Libraries/Network/FormData.js
@@ -64,10 +64,31 @@ class FormData {
     this._parts.push([key, value]);
   }
 
+  set(key, value, fileName) {
+    const newParts = [];
+    let replaced = false;
+
+    fileName = fileName || value.name || 'blob';
+
+    this._parts.forEach((part) => {
+      if (part[0] === key) {
+        newParts.push([key, value, fileName]);
+        replaced = true;
+      } else {
+        newParts.push(part);
+      }
+    });
+
+    if (!replaced) {
+      newParts.push([key, value, fileName]);
+    }
+
+    this._parts = newParts;
+  }
+
   getParts(): Array<FormDataPart> {
-    return this._parts.map(([name, value]) => {
+    return this._parts.map(([name, value, fileName]) => {
       const contentDisposition = 'form-data; name="' + name + '"';
-
       const headers: Headers = {'content-disposition': contentDisposition};
 
       // The body part is a "blob", which in React Native just means
@@ -75,13 +96,24 @@ class FormData {
       // have a `name` and `type` attribute to specify filename and
       // content type (cf. web Blob interface.)
       if (typeof value === 'object' && value) {
-        if (typeof value.name === 'string') {
-          headers['content-disposition'] += '; filename="' + value.name + '"';
-        }
-        if (typeof value.type === 'string') {
-          headers['content-type'] = value.type;
+        headers['content-disposition'] += '; filename="' + fileName + '"';
+        headers['content-type'] = value.type || 'application/octet-stream';
+
+        const blob = {
+          uri: value.uri,
+          type: value.type,
+          name: value.name,
+        };
+
+        if (value instanceof Blob) {
+          blob.uri = URL.createObjectURL(value);
         }
-        return {...value, headers, fieldName: name};
+
+        return {
+          ...blob,
+          headers,
+          fieldName: name
+        };
       }
       // Convert non-object values to strings as per FormData.append() spec
       return {string: String(value), headers, fieldName: name};
